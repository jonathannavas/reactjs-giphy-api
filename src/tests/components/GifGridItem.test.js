import React from 'react';

import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test on GifGridItem.js', () => {

    const title = 'This is a title';
    const url = 'https://localhost/hola.jpg';

    const wrapper = shallow( <GifGridItem title={title} url={url}/> );

    test('Test snapshot match', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should get a p with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('should return an image like de props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src')).toBe( url );
    });

    test('should return title data on alt attribute', () => {
        const img = wrapper.find('img');
        expect( img.prop('alt') ).toBe( title );
    });

    test('should contain animate__fadeIn', ()=>{
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toBe( true );
    });
    
});