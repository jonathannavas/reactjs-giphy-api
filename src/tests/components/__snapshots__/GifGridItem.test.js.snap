// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGridItem.js Test snapshot match 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="This is a title"
    src="https://localhost/hola.jpg"
  />
  <p>
     
    This is a title
     
  </p>
</div>
`;
